whois-api/src/main/java/net/ripe/db/whois/api/rest/ReferencesService.java

```diff
@@ -487,16 +492,12 @@ private void removeDuplicatesAndRestoreReplacedReferences(final WhoisResources w
     /**
      * Update a single object in the database
      */
-    private Response performUpdate(
+    private void performUpdate(
                 final HttpServletRequest request,
                 final RpslObject rpslObject,
                 final String deleteReason,
-                final List<String> passwords,
-                final String crowdTokenKey) {
+                final List<String> passwords, final Origin origin, final UpdateContext updateContext) {
         try {
-            final Origin origin = updatePerformer.createOrigin(request);
```

Comment:
In Delete we were calculating origin and updateContext twice.
One inside the method and the other in performUpdates. Now it is just calculated once and passing the origin and the updateContext to each method